### PR TITLE
Geant4 technical updates

### PR DESCRIPTION
**Pull request by civanch**
#### PR description:
Several changes in Geant4 simulation:
1) improved verbatim printouts on start/stop of Geant4
2) moved end of run Geant4 UI commands from master thread to worker threads
3) improved low-energy GFlash method

Mainstream results should not be changed.

#### PR validation:
private

### CODE REVIEW

**Comment by civanch** [comment]
please test

File: SimG4Core/Application/plugins/OscarMTProducer.cc [hunk 1]
```diff
@@ -103,6 +103,7 @@ namespace {
 OscarMTProducer::OscarMTProducer(edm::ParameterSet const& p, const OscarMTMasterThread* ms)
     : m_handoff{p.getUntrackedParameter<int>("workerThreadStackSize", 10 * 1024 * 1024)} {
   m_verbose = p.getParameter<int>("EventVerbose");
+  //  m_G4CommandsEndRun(p.getParameter<std::vector<std::string> >("G4CommandsEndRun")),
```
Comment (perrotta) [inline SimG4Core/Application/plugins/OscarMTProducer.cc:106]: This commented out line can get removed, as far as I can see

File: SimG4Core/Application/src/LowEnergyFastSimModel.cc [hunk 2]
```diff
@@ -68,7 +68,7 @@ void LowEnergyFastSimModel::DoIt(const G4FastTrack& fastTrack, G4FastStep& fastS
   fastStep.KillPrimaryTrack();
   fastStep.SetPrimaryTrackPathLength(0.0);
   auto track = fastTrack.GetPrimaryTrack();
-  G4double energy = track->GetKineticEnergy();
+  G4double energy = track->GetKineticEnergy() * 1.015;
```
Comment (perrotta) [inline SimG4Core/Application/src/LowEnergyFastSimModel.cc:71]: Where this factor 1.015 comes from?
Define at least a constexpr value which defines it?

**Comment by perrotta** [comment]
> 3\. improved low-energy GFlash method
> 
> Mainstream results should not be changed.

Could you please elaborate? The GFlash improvement only deals with code optimization? [This line](https://github.com/cms-sw/cmssw/pull/38542/files#diff-860a4ef4d927d366a1b0bd6915c1046284a6eb81db367ffab11bcdd9d4c04cc6R71) makes me think the opposite...
And by the way, which are the "non mainstream" results which get changed instead

**Comment by civanch** [comment]
@perrotta , thanks for comments, will address them. Concerning "mainstream": this GFlash method can be used if is enabled, it was used for private validations only (where from 1.015 appears), we plan introduce this method in 13_0 after validation.

**Comment by perrotta** [comment]
please test

**Pull request merged**